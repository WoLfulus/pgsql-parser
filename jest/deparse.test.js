const parser = require('../src');

const json = require(__dirname + '/fixtures/policies.json');

describe('extensions', () => {
  it('deparse', () => {
    const sql = parser.deparse(json);
    expect(sql.trim()).toMatchSnapshot();
  });
});
