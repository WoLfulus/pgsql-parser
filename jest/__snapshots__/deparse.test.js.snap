// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extensions deparse 1`] = `
"BEGIN;

CREATE EXTENSION IF NOT EXISTS \\"blah\\" CASCADE;

GRANT USAGE ON SCHEMA users TO \\"administrator\\";

GRANT EXECUTE ON FUNCTION \\"auth\\".\\"authenticate\\" TO \\"anonymous\\";

GRANT SELECT, INSERT, UPDATE, DELETE ON TABLE \\"auth\\" . \\"token\\" TO \\"administrator\\";

CREATE POLICY \\"delete_user\\" ON \\"users\\" . \\"user\\" FOR DELETE TO \\"authenticated\\" USING ((\\"id\\") = (current_setting('user.id')::uuid));

ROLLBACK;"
`;
